src/modules/sensors/sensors.cpp

```diff
@@ -2215,15 +2221,21 @@ Sensors::task_main()
 		mag_poll(raw);
 		baro_poll(raw);
 
-		/* work out if main gyro timed out and fail over to alternate gyro */
+		/* work out if main gyro timed out and fail over to alternate gyro, if available */
 		if (hrt_elapsed_time(&raw.gyro_timestamp[0]) > 20 * 1000) {
+			warnx("gyro has timed out");
 
-			/* if the secondary failed as well, go to the tertiary */
+			/* if the secondary failed as well, go to the tertiary, if available */
 			if (hrt_elapsed_time(&raw.gyro_timestamp[1]) > 20 * 1000) {
-				fds[0].fd = _gyro_sub[2];
-
+				if (_gyro_sub[2] != -1) {
+					warnx("using tertiary gyro");
+					fds[0].fd = _gyro_sub[2];
+				}
 			} else {
-				fds[0].fd = _gyro_sub[1];
+				if (_gyro_sub[1] != -1) {
+					warnx("using secondary gyro");
+					fds[0].fd = _gyro_sub[1];
+				}
```

Comment:
There should be some indication that main, tertiary and secondary failed or were not available.